security/hcaptcha.go

```diff
@@ -0,0 +1,85 @@
+package security
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type CaptchaResponse struct {
+	Token string `json:"hcaptchaToken"`
```

Comment:
nit: we use snake_case instead of camelCase in gotrue for json fields

Reply:
should we also enforce users to pass the site_key as well? seems like it's recommended on the hcaptcha docs (https://docs.hcaptcha.com/#verify-the-user-response-server-side)

Reply:
we can also get the client's ip address using `r.RemoteAddr` and pass it in the POST request to hcaptcha too

Reply:
Renamed to  use snake case, and started passing client IP along. I've intentionally skipped site key for now; it's not necessary to get value out of this chagne, and I'm not sure what we want support for that to look like (e.g. keyed on the origin to support multiple keys? some other metric?)

Reply:
(skipping abstracting out things to support other providers for the same reason)